### PR TITLE
add User Preferences

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Add User Preferences.

### Which issue does this PR related to?
fixed https://github.com/eclipse/che/issues/18526
fixed https://github.com/eclipse/che/issues/18867

Screenshot from 2021-01-18 02-27-51

Screenshot from 2021-01-25 16-15-59
Screenshot from 2021-01-25 16-14-59
Screenshot from 2021-01-25 16-15-25
Screenshot from 2021-01-25 16-22-11

Signed-off-by: Oleksii Orel <oorel@redhat.com>

### CODE REVIEW

File: src/pages/UserPreferences/index.tsx [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    ];
+
+    return (
+      <React.Fragment>
+        <Head pageName="User Preferences" />
+        <PageSection variant={PageSectionVariants.light}>
+          <Title headingLevel={'h1'}>User Preferences</Title>
+          <TextContent>
+            <Text component={TextVariants.p}>
+              Control the configuration and policies for your installation.
```
Comment (sleshchenko) [inline src/pages/UserPreferences/index.tsx:233]: It seems to be outdated description but I don't have a good suggestion on how it should look like. On the suggested design of external accounts, it's just placeholder.

Can we just remove it until we don't have a good idea what should be here?

@sympatheticmoose Do you have an idea on that part?

**Review by sleshchenko — COMMENTED** [review]
Seems changing password or username does not enable `Save` button.

File: src/pages/UserPreferences/index.tsx (hunk 1 above)
Comment (sympatheticmoose) [reply src/pages/UserPreferences/index.tsx:233]: @sleshchenko Remove seems sensible for now whilst it's just the single tab. Curious why we've switched from Container Registry/Registries to Docker?

Comment (sleshchenko) [reply src/pages/UserPreferences/index.tsx:233]: We didn't! It's just a typo :-D

@olexii4 please fix it

Comment (olexii4) [reply src/pages/UserPreferences/index.tsx]: I have fixed it.

**Comment by olexii4** [comment]
@sleshchenko I have fixed a bug when changing password or username does not enable the ```Save``` button.

File: src/pages/UserPreferences/index.tsx (hunk 1 above)
Comment (sympatheticmoose) [reply src/pages/UserPreferences/index.tsx]: @olexii4 just checking you also saw the comment that we should be referring to Container Registry / Registries rather than Docker?

Comment (olexii4) [reply src/pages/UserPreferences/index.tsx]: @sympatheticmoose  I have fixed it.

Comment (sympatheticmoose) [reply src/pages/UserPreferences/index.tsx]: Awesome. Thanks @olexii4

**Review by JPinkney — COMMENTED** [review]
I just have a couple minor things to comment on. I wonder if we should change all instances of `dockerCredentials` in the code to something like `containerCredentials` to keep it more inline with the UI.

Also, I ran into a bug where you can click the registry and then a blank screen pops up

potential-user-preferences-bug

**Comment by olexii4** [comment]
@JPinkney I have fixed the name 'dockerCredentials' and want to note that we still have the object  { dockerCredentials: '...' }  for backward compatibility(existing UD).
Screenshot from 2021-01-19 05-26-38

File: webpack.config.prod-dev.js [hunk 2]
```diff
@@ -46,14 +46,6 @@ module.exports = env => {
             origin: proxyTarget
           },
         },
-        '/workspace-loader': {
```
Comment (JPinkney) [inline webpack.config.prod-dev.js:49]: Is this a related change to the PR?

**Review by JPinkney — APPROVED** [review]
Tested again and my issue is now fixed (it now brings you to the registry url if you click the row). My only suggestion would be if you click the row should the registry be opened in a new tab? I also just have one other minor comment but I'm going to approve anyways

File: webpack.config.prod-dev.js (hunk 2 above)
Comment (olexii4) [reply webpack.config.prod-dev.js:49]: No. This is not a related change to the PR. I will remove it.

**Comment by olexii4** [comment]
@JPinkney I have changed the  flow for check the registry URL(opened in a new tab):
Screenshot from 2021-01-25 16-15-59
Screenshot from 2021-01-25 16-14-59
Screenshot from 2021-01-25 16-15-25

File: src/pages/UserPreferences/Modals/EditRegistryModal.tsx [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+import { RegistryUsernameFormGroup } from '../RegistryUsername';
+
+type Props = {
+  registries: RegistryRow[];
+  currentIndex: number;
+  isOpen: boolean;
+  onChange: (registries: RegistryRow[], isEditMode: boolean) => void;
+  onCancel: () => void;
+}
+// type Props = {
```
Comment (JPinkney) [inline src/pages/UserPreferences/Modals/EditRegistryModal.tsx:34]: I think this might be leftover

Comment (olexii4) [reply src/pages/UserPreferences/Modals/EditRegistryModal.tsx]: I have fixed it

**Review by akurinnoy — APPROVED** [review]
